fmt subcommand includes recursive flag even on older versions

I am trying to run fmt with terraform 0.11.7 but keep getting the usage method and the fmt command fails. I added the TF_LOG: 'trace' option to the env vars and found this.

```
2019/11/01 15:55:11 [INFO] CLI command args: []string{"fmt", "-check=true", "-list=true", "-check", "-write=false", "-list", "-recursive"}
flag provided but not defined: -recursive
Usage: terraform fmt [options] [DIR]

	Rewrites all Terraform configuration files to a canonical format.

	If DIR is not specified then the current working directory will be used.
	If DIR is "-" then content will be read from STDIN.

Options:

  -list=true       List files whose formatting differs (always false if using STDIN)

  -write=true      Write result to source file instead of STDOUT (always false if using STDIN or -check)

  -diff=false      Display diffs of formatting changes

  -check=false     Check if the input is formatted. Exit status will be 0 if all input is properly formatted and non-zero otherwise.
2019/11/01 15:55:11 [DEBUG] plugin: waiting for all plugin processes to complete...

##[error]Docker run failed with exit code 1
```


This is my code block

```
      - name: 'Terraform Format'                                            
        uses: hashicorp/terraform-github-actions@master                     
        with:                                                               
          tf_actions_version: 0.11.7                                        
          tf_actions_subcommand: 'fmt'                                      
          tf_actions_working_dir: 'infrastructure'                          
          tf_actions_comment: false                                         
        env:                                                                                                   
          TF_CLI_ARGS: '-check=true -list=true'                             
          TF_LOG: 'TRACE'
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## v0.5.1
+
+### Fixed
+
+* Do not use `-recursive` option with `terraform fmt` for Terraform 0.11.x. ([#90](https://github.com/hashicorp/terraform-github-actions/issues/90))
+
 ## v0.5.0
 
 ### Added
@@ -9,10 +15,6 @@
 ### Changed
 
 * Completely refactored the codebase into one GitHub Action. Please refer to the README for current usage.
-
-### Deprecated
-
-N/A
 
 ### Removed
 
@@ -25,7 +27,3 @@ N/A
 * Added support for Terraform 0.11.14. ([#42](https://github.com/hashicorp/terraform-github-actions/issues/67))
 * Comments will not be posted to pull requests when `terraform plan` contains no changes. ([#29](https://github.com/hashicorp/terraform-github-actions/issues/67))
 * Added ability to specify a Terraform version to use. ([#23](https://github.com/hashicorp/terraform-github-actions/issues/67))
-
-### Security
-
-N/A

--- a/src/main.sh
+++ b/src/main.sh
@@ -4,6 +4,17 @@ function stripColors {
   echo "${1}" | sed 's/\x1b\[[0-9;]*m//g'
 }
 
+function hasPrefix {
+  case ${2} in
+    "${1}"*)
+      true
+      ;;
+    *)
+      false
+      ;;
+  esac
+}
+
 function parseInputs {
   # Required inputs
   if [ "${INPUT_TF_ACTIONS_VERSION}" != "" ]; then

--- a/src/terraform_fmt.sh
+++ b/src/terraform_fmt.sh
@@ -1,9 +1,15 @@
 #!/bin/bash
 
 function terraformFmt {
+  # Eliminate `-recursive` option for Terraform 0.11.x.
+  fmtRecursive="-recursive"
+  if hasPrefix "0.11" "${tfVersion}"; then
+    fmtRecursive=""
+  fi
+
   # Gather the output of `terraform fmt`.
   echo "fmt: info: checking if Terraform files in ${tfWorkingDir} are correctly formatted"
-  fmtOutput=$(terraform fmt -check -write=false -diff -recursive 2>&1)
+  fmtOutput=$(terraform fmt -check -write=false -diff ${fmtRecursive} 2>&1)
   fmtExitCode=${?}
   
   # Exit code of 0 indicates success. Print the output and exit.

